feature/gnoi/system/tests/per_component_reboot_test/per_component_reboot_test.go

```diff
@@ -215,3 +216,81 @@ func TestLinecardReboot(t *testing.T) {
 	helpers.ValidateOperStatusUPIntfs(t, dut, intfsOperStatusUPBeforeReboot, 10*time.Minute)
 	// TODO: Check the line card uptime has been reset.
 }
+
+// Reboot the fabric component on the DUT.
+func TestFabricReboot(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	if deviations.GNOIFabricComponentRebootUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation deviation_gnoi_fabric_component_reboot_unsupported")
+	}
+
+	const fabricBootTime = 10 * time.Minute
+	fabrics := components.FindComponentsByType(t, dut, fabricType)
+	t.Logf("Found fabric components: %v", fabrics)
+
+	t.Logf("Find a removable fabric component to reboot.")
+	var removableFabric string
+	for _, fabric := range fabrics {
+		t.Logf("Check if %s is removable", fabric)
+		if removable, ok := gnmi.Lookup(t, dut, gnmi.OC().Component(fabric).Removable().State()).Val(); ok {
+			if removable {
+				t.Logf("Found removable fabric component: %v", fabric)
+				removableFabric = fabric
+			} else {
+				t.Logf("Found non-removable fabric component: %v", fabric)
+			}
+		} else {
+			t.Logf("Detected non-removable fabric component: %v", fabric)
+		}
+	}
+	if removableFabric == "" {
+		t.Fatalf("Component(fabric).Removable().Get(t): got none, want non-empty")
+	}
+
+	// Fetch list of interfaces which are up prior to fabric component reboot.
+	intfsOperStatusUPBeforeReboot := helpers.FetchOperStatusUPIntfs(t, dut, *args.CheckInterfacesInBinding)
+	t.Logf("OperStatusUP interfaces before reboot: %v", intfsOperStatusUPBeforeReboot)
+
+	// Fetch a new gnoi client.
+	gnoiClient := dut.RawAPIs().GNOI().Default(t)
+	useNameOnly := deviations.GNOISubcomponentPath(dut)
+	rebootSubComponentRequest := &spb.RebootRequest{
+		Method: spb.RebootMethod_COLD,
+		Subcomponents: []*tpb.Path{
+			components.GetSubcomponentPath(removableFabric, useNameOnly),
+		},
+	}
+
+	t.Logf("rebootSubComponentRequest: %v", rebootSubComponentRequest)
+	rebootResponse, err := gnoiClient.System().Reboot(context.Background(), rebootSubComponentRequest)
+	if err != nil {
+		t.Fatalf("Failed to perform fabric component reboot with unexpected err: %v", err)
+	}
+	t.Logf("gnoiClient.System().Reboot() response: %v, err: %v", rebootResponse, err)
+
+	rebootDeadline := time.Now().Add(fabricBootTime)
+	for retry := true; retry; {
+		t.Log("Waiting for 10 seconds before checking.")
+		time.Sleep(10 * time.Second)
+		if time.Now().After(rebootDeadline) {
+			retry = false
+			break
```

Comment:
because you break out of the loop, there's no need to set retry to false. I think the retry variable doesn't quite pull it's weight here. Instead the loop can be:

```
for {
	t.Log("Waiting for 10 seconds before checking.")
	time.Sleep(10 * time.Second)
	if time.Now().After(rebootDeadline) {
		break
	}
	resp, err := gnoiClient.System().RebootStatus(context.Background(), &spb.RebootStatusRequest{})
	if status.Code(err) == codes.Unimplemented {
		t.Fatalf("Unimplemented RebootStatus() is not fully compliant with the Reboot spec.")
	}
	if !resp.GetActive() {
		break
	}
}
```

Reply:
Updated the loop